controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -430,3 +431,35 @@ func nilNodeRef(machine clusterv1.Machine) clusterv1.Machine {
 	machine.Status.NodeRef = nil
 	return machine
 }
+
+func TestRemoveMemberForNode(t *testing.T) {
+	t.Run("do not remove the etcd member if the cluster has fewer than 2 control plane nodes", func(t *testing.T) {
+		expectedErr := errors.New("cluster has fewer than 2 control plane nodes; removing an etcd member is not supported")
+
+		workloadCluster := &Cluster{
+			Client: &fakeClient{
+				list: &corev1.NodeList{
+					Items: []corev1.Node{
+						nodeNamed("first-control-plane"),
+					},
+				},
+			},
+		}
+
+		err := workloadCluster.removeMemberForNode(context.Background(), "first-control-plane")
+		if err == nil || errors.Is(err, expectedErr) {
```

Comment:
`errors.Is` doesn't do a string comparison so this will always return false which is why this test is passing. Because the test has complex logic (I'm counting `||` as complex) it's easy to make the test do something it shouldn't be which is why I am arguing above to have the test named specifically to check for this particular error.

To see `errors.Is` in action, please see this play: https://play.golang.org/p/QuZUMQ9u1Ol

To simplify the logic in this test, one alternative would be to first check to make sure that err != nil. If err != nil, then this test can fatal since there is a nil err but we aren't expecting one.

After that, a check that the error is the expected error. However to make this work, the expected err needs to be a package level var and not simply equivalent strings (it could also be a full fledged type, but we don't need that complexity just yet). This also aids in test longevity because the error string is only defined in one place instead of two.

Reply:
Thanks for the explanation! Will address both points.